File: bench/memory/3.cpp
#include <iostream>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	int S = 1000000; // 1 MB
	int N = 1000;
	if (argc >= 2)
	{
		S = atoi(argv[1]);
		if (argc == 3)
		{
			N = atoi(argv[2]);
		}
	}

	S = S / sizeof(int);

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < N; i++)
	{
		int *p = (int *)malloc(S * sizeof(int));
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}

File: bench/memory/3.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var S = 1000000 // 1 MB
	var N = 1000
	var args = os.Args[1:]
	if len(args) >= 1 {
		S, _ = strconv.Atoi(args[0])
		if len(args) >= 2 {
			N, _ = strconv.Atoi(args[1])
		}
	}

	S = S / strconv.IntSize

	var start = time.Now()

	for i := 0; i < N; i++ {
		_ = make([]int, S)
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}
